build-logic/build.gradle.kts

```diff
@@ -1,12 +1,34 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
 plugins {
    `kotlin-dsl`
+   kotlin("jvm") version embeddedKotlinVersion
 }
 
-repositories {
-   gradlePluginPortal()
+dependencies {
+   implementation(platform(libs.kotlin.bom))
+   implementation(libs.gradlePlugin.kotlin)
+   implementation(libs.gradlePlugin.kotlinSerialization)
+   implementation(libs.gradlePlugin.testlogger)
+   implementation(libs.gradlePlugin.kotest)
 }
 
-dependencies {
-   implementation(libs.kotlin.gradle.plugin)
-   implementation(libs.testlogger.gradle.plugin)
+val buildLogicJvmVersion = "11"
+val buildLogicJvmTarget = "1.8"
+
+
+kotlin {
+   jvmToolchain {
+      (this as JavaToolchainSpec).languageVersion.set(JavaLanguageVersion.of(buildLogicJvmVersion))
+   }
+}
+
+tasks.withType<KotlinCompile>().configureEach {
+   kotlinOptions {
+      jvmTarget = buildLogicJvmTarget
+   }
+}
+
+kotlinDslPluginOptions {
+   jvmTarget.set(buildLogicJvmTarget)
```

Comment:
Why do we configure this here and not in the conventions plugin?

Reply:
Chicken and egg :) we'd have to make another included-build that would provide plugins for the build-logic project, and that's a lot more complication that's not needed

Reply:
Ah, gotcha. 😄